components/raftstore/src/store/fsm/peer.rs

```diff
@@ -6679,6 +6710,41 @@ where
         self.schedule_tick(PeerTick::ReportBuckets)
     }
 
+    /// Check whether the peer should send a request to fetch the committed
+    /// index from the leader.
+    fn try_to_fetch_committed_index(&mut self) {
+        // Already completed, skip.
+        if !self.fsm.peer.needs_update_last_leader_committed_idx() {
+            return;
+        }
+        // Construct a MsgReadIndex message and send it to the leader to
+        // fetch the latest committed index of this raft group.
+        let leader_id = self.fsm.peer.leader_id();
+        if leader_id == raft::INVALID_ID {
+            // The leader is unknown, so we can't fetch the committed index.
+            return;
+        }
+        let mut msg = raft::eraftpb::Message::new();
+        msg.set_msg_type(MessageType::MsgReadIndex);
+        let rctx = ReadIndexContext {
+            id: uuid::Uuid::new_v4(),
+            request: None,
+            locked: None,
+        };
+        let mut e = raft::eraftpb::Entry::default();
+        e.set_data(rctx.to_bytes().into());
+        msg.mut_entries().push(e);
+        msg.from = self.fsm.peer_id();
+        msg.to = leader_id;
+        let raft_msg = self.fsm.peer.build_raft_messages(self.ctx, vec![msg]);
+        self.fsm.peer.send_raft_messages(self.ctx, raft_msg);
```

Comment:
How about the following?

```suggestion
        let rctx = ReadIndexContext {
            id: uuid::Uuid::new_v4(),
            request: None,
            locked: None,
        };
        self.fsm.peer.raft_group.read_index(rctx.to_bytes());
```